crates/sui-core/src/checkpoints/checkpoint_executor/mod.rs

```diff
@@ -412,26 +412,47 @@ impl CheckpointExecutor {
             }
         }
 
+        let exec_start = Instant::now();
+        let prepare_elapsed = exec_start - prepare_start;
+        self.metrics
+            .checkpoint_prepare_latency_us
+            .report(prepare_elapsed.as_micros() as u64);
+        if checkpoint.sequence_number % CHECKPOINT_PROGRESS_LOG_COUNT_INTERVAL == 0 {
+            info!(
+                "Checkpoint preparation for execution took {:?}",
+                prepare_elapsed
+            );
```

Comment:
and here. I see that we're only logging some of these every 5k checkpoints, but only for `info` level logs. Seems like it will become confounded in debug mode without explicitly logging the checkpoint sequence number.

Reply:
This function `execute_transactions()` is called by `schedule_checkpoint()` indirectly, which annotates the log with sequence num and epoch.